refactor: replace atomic counters with sliding window in circuit breaker

aviyam181199 opened the pull request:
Improves time-based failure tracking accuracy by using a sliding window with progressive bucket rotation instead of atomic counters with instant resets. Provides better thread-safety, eliminates race conditions, and ensures consistent behavior across count-based and ratio-based modes.

Adapted from stale PR #1020.

aviyam181199 committed:
refactor: replace atomic counters with sliding window in circuit breaker

Improves time-based failure tracking accuracy by using a sliding window with progressive bucket rotation instead of atomic counters with instant resets. Provides better thread-safety, eliminates race conditions, and ensures consistent behavior across count-based and ratio-based modes.

jeevatkm reviewed (changes requested):
@aviyam181199 Thanks for your work on state PR. I’ve just merged the circuit breaker hooks PR. 
Could you please resolve any PR conflicts and ensure that the test case coverage is complete?

aviyam181199 committed:
test: add SetInterval coverage for sliding window

Adds test case for SetInterval method to achieve 100% test coverage on circuit breaker sliding window implementation.

jeevatkm reviewed (approved):
Thanks @aviyam181199 for the test coverage and working on the stale PR. 

Thanks @liuzengh for the PR.
